packages/lodestar/src/db/controller/impl/level.ts

```diff
@@ -86,7 +86,7 @@ export class LevelDbController extends EventEmitter implements IDatabaseControll
         gt: opts.gt,
         lt: opts.lt,
       }).on("data", function (data) {
-        searchData.push(data);
+        searchData.push(data.filter((datum, index) => index % (opts.step || 1) === 0));
```

Comment:
ehm, pretty sure data is one item from database like, that function is called multiple times.
You should count how many times this method is called I guess

Reply:
Yeah, data is a single item. You'll want to wrap the `searchData.push(data)` in an `if` that does the proper check. You may need to use a [`createReadStream`](https://github.com/Level/levelup#dbcreatereadstreamoptions) so you have access to the keys that you want to filter by.

Reply:
Are the keys array indices or block heights, though? Like can I assume that some key is equal to the last key plus one?

Reply:
I don't think thats a safe assumption. There may be skip slots, where we only store eg: block 2, 3, 5

Reply:
Gotcha. And just to be sure, if a client calls this rpc method with a start slot 2, count 3, and step 2, do we return [2, 3, 5] or [2, 3]?

Reply:
I would say 2,3,5 as you he requested 3 blocks

Reply:
Sounds good. Thanks for the clarification y'all.